client/asset/dcr/rpcwallet.go

```diff
@@ -245,8 +255,20 @@ func (w *rpcWallet) checkRPCConnection(ctx context.Context) error {
 // Connect establishes a connection to the previously created rpc client.
 // Part of the Wallet interface.
 func (w *rpcWallet) Connect(ctx context.Context) error {
+	w.rpcMtx.Lock()
+	defer w.rpcMtx.Unlock()
+
```

Comment:
After successful `Connect`, is `rpcReadyToConnect` still `true`? So a subsequent call to `Connect` without a call to `Disconnect` would make it to `rpcConnector.Connect` for a second time on the same `rpcConnector`. In theory, the caller would never call `Connect` again before calling `Disconnect`. We can work under that assumption. I just wanted to make sure I'm understanding the intent.

Reply:
My thinking here is that I'd rather have a second (incorrect) call to `Connect` with no prior `Disconnect` to be met with an "already connected" error than letting the rpcclient get recreated.  But yeah, never call `Connect` again before calling `Disconnect` is the idea.